fdbserver/VersionedBTree.actor.cpp

```diff
@@ -5152,10 +5164,7 @@ class VersionedBTree {
 			m_keyProvider = makeReference<NullKeyProvider>();
 		}
 
-		m_pBoundaryVerifier = DecodeBoundaryVerifier::getVerifier(name);
-		if (m_pBoundaryVerifier != nullptr) {
-			m_pBoundaryVerifier->setKeyProvider(m_keyProvider);
-		}
+		DecodeBoundaryVerifier::setKeyProvider(name, m_keyProvider);
```

Comment:
We need to keep the `m_pBoundaryVerifier = DecodeBoundaryVerifier::getVerifier(name);` line though. It is used later by Redwood to verify if the B-tree has corruption.